Orange/widgets/data/owdatasets.py

```diff
@@ -208,71 +200,64 @@ def __init__(self):
+        )
+        self.view.setItemDelegateForColumn(
+            self.Header.size,
+            Orange.widgets.data.owdatasets.SizeDelegate(self))
 
-        allinforemote = res  # type: Dict[Tuple[str, ...], dict]
-        allkeys = set(allinfolocal)
-        if allinforemote is not None:
-            allkeys = allkeys | set(allinforemote)
-        allkeys = sorted(allkeys)
+        self.view.setItemDelegateForColumn(
+            self.Header.instances,
+            Orange.widgets.data.owdatasets.NumericalDelegate(self)
+        )
+        self.view.setItemDelegateForColumn(
+            self.Header.variables,
+            Orange.widgets.data.owdatasets.NumericalDelegate(self)
+        )
 
-        def info(file_path):
-            if file_path in allinforemote:
-                info = allinforemote[file_path]
-            else:
-                info = allinfolocal[file_path]
-            islocal = file_path in allinfolocal
-            isremote = file_path in allinforemote
-            outdated = islocal and isremote and (
-                allinforemote[file_path].get('version', '') !=
-                allinfolocal[file_path].get('version', ''))
-            islocal &= not outdated
-            prefix = os.path.join('', *file_path[:-1])
-            filename = file_path[-1]
-
-            return namespace(
-                file_path=file_path,
-                prefix=prefix, filename=filename,
-                title=info.get("title", filename),
-                datetime=info.get("datetime", None),
-                description=info.get("description", None),
-                references=info.get("references", []),
-                seealso=info.get("seealso", []),
-                source=info.get("source", None),
-                year=info.get("year", None),
-                instances=info.get("instances", None),
-                variables=info.get("variables", None),
-                target=info.get("target", None),
-                missing=info.get("missing", None),
-                tags=info.get("tags", []),
-                size=info.get("size", None),
-                islocal=islocal,
-                outdated=outdated
-            )
+    def _parse_info(self, file_path):
+        if file_path in self.allinfo_remote:
+            info = self.allinfo_remote[file_path]
+        else:
+            info = self.allinfo_local[file_path]
+
+        islocal = file_path in self.allinfo_local
+        isremote = file_path in self.allinfo_remote
+
+        outdated = islocal and isremote and (
+                self.allinfo_remote[file_path].get('version', '') != self.allinfo_local[file_path].get('version', '')
+        )
+        islocal &= not outdated
+
+        prefix = os.path.join('', *file_path[:-1])
+        filename = file_path[-1]
+
+        # add missing variables to the namespace
+        # not needed if all info files have the same JSON structure
+        for tag in self._html_description_tags:
```

Comment:
Not all variables are in `_html_description_tags`, e.g. one test fails, because 'size' is missing (which is in `_header_index`, but not in `_html_description_tags`.
Also, did you make sure that it is OK to set all missing things to `None`? Previously, some default values were different, e.g. `[]` for 'references'.